Code/Framework/AzToolsFramework/AzToolsFramework/ComponentModes/BoxViewportEdit.cpp

```diff
@@ -34,23 +30,59 @@ namespace AzToolsFramework
     {
     }
 
-    void BoxViewportEdit::UpdateManipulators()
+    void BoxViewportEdit::InstallGetBoxDimensions(const AZStd::function<AZ::Vector3()>& getBoxDimensions)
     {
-        AZ::Transform manipulatorSpace = AZ::Transform::CreateIdentity();
-        ShapeManipulatorRequestBus::EventResult(
-            manipulatorSpace, m_entityComponentIdPair, &ShapeManipulatorRequestBus::Events::GetManipulatorSpace);
+        m_getBoxDimensions = getBoxDimensions;
+    }
+
+    void BoxViewportEdit::InstallGetLocalTransform(const AZStd::function<AZ::Transform()>& getLocalTransform)
+    {
+        m_getLocalTransform = getLocalTransform;
+    }
 
-        AZ::Vector3 nonUniformScale = AZ::Vector3::CreateOne();
-        AZ::NonUniformScaleRequestBus::EventResult(
-            nonUniformScale, m_entityComponentIdPair.GetEntityId(), &AZ::NonUniformScaleRequestBus::Events::GetScale);
+    void BoxViewportEdit::InstallSetBoxDimensions(const AZStd::function<void(const AZ::Vector3)>& setBoxDimensions)
+    {
+        m_setBoxDimensions = setBoxDimensions;
+    }
 
-        AZ::Vector3 boxDimensions = AZ::Vector3::CreateZero();
-        BoxManipulatorRequestBus::EventResult(
-            boxDimensions, m_entityComponentIdPair, &BoxManipulatorRequestBus::Events::GetDimensions);
+    AZ::Vector3 BoxViewportEdit::GetBoxDimensions() const
+    {
+        if (m_getBoxDimensions)
+        {
+            return m_getBoxDimensions();
+        }
+        AZ_WarningOnce("BoxViewportEdit", false, "No implementation provided for GetBoxDimensions");
```

Comment:
Are the 3 Install functions a requirement? I see there are these `AZ_WarningOnce` when they are not setup, is this going to break functionality? is it a possible use case to not set all the install functions? If yes, then maybe there is no need for a warning, if no then maybe this could be an error since the class is not setup properly? I wonder if a warning once would just be missed. Can you give more info about this and why warning once is enough?

Reply:
I made it only fire once to avoid spam, but it probably makes sense to upgrade it to AZ_ErrorOnce. I think you will always need all of those functions installed, or the manipulators won't work correctly.